File: users-service/src/main/java/net/plshark/auth/service/TokenVerifier.java
package net.plshark.auth.service;

import reactor.core.publisher.Mono;

/**
 * Verifies JWTs
 */
public interface TokenVerifier {

    /**
     * Verify an access token
     * @param token the token
     * @return the username of the corresponding user if successful or a BadCredentialsException if the token is invalid
     * or expired
     */
    Mono<String> verifyToken(String token);

    /**
     * Verify a refresh token
     * @param token the token to verify
     * @return the username if successful or a BadCredentialsException if the token is invalid or expired
     */
    Mono<String> verifyRefreshToken(String token);
}

File: user-error-server/src/main/kotlin/net/plshark/usererror/authentication/token/TokenVerifier.kt
package net.plshark.usererror.authentication.token

import net.plshark.usererror.authorization.AuthenticatedUser

/**
 * Verifies JWTs
 */
interface TokenVerifier {

    /**
     * Verify an access token
     * @param token the token
     * @return the username and authorities of the corresponding user if successful
     * @throws BadCredentialsException if the token is invalid or expired
     */
    fun verifyToken(token: String): AuthenticatedUser

    /**
     * Verify a refresh token
     * @param token the token to verify
     * @return the username if successful
     * @throws BadCredentialsException if the token is invalid or expired
     */
    fun verifyRefreshToken(token: String): String
}